### PR TITLE
fix: detect parameters without schema or content fields

### LINKED ISSUE #568: Parameters without schema or content fields
**Describe the bug**

OpenAPI CLI should detect `Parameters` without `schema` or `content` fields while the OpenAPI specification states that one of these fields MUST be specified. This should be part of the `spec` rule.

**To Reproduce**
TBD

**Expected behavior**
Show a problem if none of schema or content are defined or if both schema and content are defined.

**OpenAPI definition**
3.0 and 3.1

**`openapi-cli` Version(s)**
1.1.0-beta.83

**`Node.js` Version(s)**
14

**Additional context**

This came up in a slack discussion.

### LINKED ISSUE #570: Request for Schema validation
**Is your feature request related to a problem? Please describe.**
No - This is not related to any problems.

It seems like currently, we don't have any schema validation for request parameters. I think it would be great if you could add a schema evaluation like the online swagger editor.

**Describe the solution you'd like**

If we do not have a scheme for the request parameters, I think we can raise an error as shown below. 

image

**OpenAPI Specification**

```yaml
openapi: 3.0.0
info:
  title: Petstore API
  version: 1.0.0
  description: description

paths:

  /v1/pets/{petId}:
    get:
      tags:
        - pets
      summary: Gets a specific pet
      parameters:
      - name: petId
        in: path
      - name: limit
        in: query
      responses:
        '200':
          description: Details of the requested pet.
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Pet'

components:

  schemas:
    Pet:
      type: object
      properties:
        resourceType:
          type: string
```

### PR DESCRIPTION
**Pull request by AntonKozachuk**
## What/Why/How?
OpenAPI CLI should detect Parameters without schema or content fields while the OpenAPI specification states that one of these fields MUST be specified.
image

## Reference
Fixes #568 
Fixes #570 

## Testing
Tested locally

## Screenshots (optional)
N/a

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/rules/common/spec.ts [hunk 1]
```diff
@@ -37,6 +37,18 @@ export const OasSpec: Oas3Rule | Oas2Rule = () => {
         }
       }
 
+      if (oasVersion !== 'oas2') {
```
Comment (RomanHotsiy) [inline packages/core/src/rules/common/spec.ts:40]: Better to add something like `oneOfProperty` into the type tree here https://github.com/Redocly/openapi-cli/blob/master/packages%2Fcore%2Fsrc%2Ftypes%2Foas3.ts#L121

**Review by RomanHotsiy — APPROVED** [review]

File: packages/core/src/types/oas3.ts [hunk 2]
```diff
@@ -136,6 +136,7 @@ const Parameter: NodeType = {
     content: 'MediaTypeMap',
   },
   required: ['name', 'in'],
+  requiredOneOf: ['schema', 'content']
```
Comment (Marshevskyy) [inline packages/core/src/types/oas3.ts:139]: would it work with OAS 3.1?

Comment (AntonKozachuk) [reply packages/core/src/types/oas3.ts:139]: Yes, because Oas3_1Types include Oas3Types (and Parameter type too):

Comment (adamaltman) [reply packages/core/src/types/oas3.ts:139]: Do we have a code style for a trailing comma on these? I notice, for example, a trailing comma on the line above.

**Review by tatomyr — APPROVED** [review]

File: packages/core/src/types/oas3.ts (hunk 2 above)
Comment (AntonKozachuk) [reply packages/core/src/types/oas3.ts:139]: Not sure.

Comment (AntonKozachuk) [reply packages/core/src/types/oas3.ts:139]: @adamaltman can I merge this or hold it till we found a common ground on commas?

Comment (RomanHotsiy) [reply packages/core/src/types/oas3.ts:139]: We do have but looks like this project does not have prettier set up. @tatomyr would be great to do it on the cool-down.

Comment (RomanHotsiy) [inline packages/core/src/types/oas3.ts:139]:
```suggestion
  requiredOneOf: ['schema', 'content'],
```

Comment (RomanHotsiy) [reply packages/core/src/types/oas3.ts]: @AntonKozachuk I added the comma 😄

**Pull request merged**